Kernel/Locking/Mutex.cpp

```diff
@@ -207,19 +208,31 @@ void Mutex::block(Thread& current_thread, Mode mode, SpinlockLocker<Spinlock>& l
     if constexpr (LOCK_IN_CRITICAL_DEBUG)
         VERIFY_INTERRUPTS_ENABLED();
     m_blocked_thread_lists.with([&](auto& lists) {
-        auto& list = lists.list_for_mode(mode);
-        VERIFY(!list.contains(current_thread));
-        list.append(current_thread);
+        auto append_to_list = [&]<typename L>(L& list) {
+            VERIFY(!list.contains(current_thread));
+            list.append(current_thread);
+        };
+
+        if (m_behavior == MutexBehavior::BigLock)
+            append_to_list(lists.exclusive_big_lock);
+        else
+            append_to_list(lists.list_for_mode(mode));
     });
 
     dbgln_if(LOCK_TRACE_DEBUG, "Mutex::lock @ {} ({}) waiting...", this, m_name);
     current_thread.block(*this, lock, requested_locks);
     dbgln_if(LOCK_TRACE_DEBUG, "Mutex::lock @ {} ({}) waited", this, m_name);
 
     m_blocked_thread_lists.with([&](auto& lists) {
-        auto& list = lists.list_for_mode(mode);
-        VERIFY(list.contains(current_thread));
-        list.remove(current_thread);
+        auto remove_from_list = [&]<typename L>(L& list) {
+            VERIFY(list.contains(current_thread));
+            list.remove(current_thread);
+        };
+
+        if (m_behavior == MutexBehavior::BigLock)
+            remove_from_list(lists.exclusive_big_lock);
```

Comment:
Don't we need to also update `Mutex::unblock_waiters` to look at the `exclusive_big_lock` list.
It looks like if someone blocks on the big process lock at the moment, they would never be woken up, as no one would know how to find their thread?

Reply:
That should've been obvious... thanks @bgianfo! Pushed some new changes.